### DISCUSSION TITLE
Custom array component

### DISCUSSION THREAD

**Discussion by Zetra416** [vazco/uniforms#1196] [Q&A] [answered]
hi continue from my previous question about using array type field, i have another question, so i have this kind of schema in my code with jsonschema

```js
(() => {
  const ajv = new Ajv({ allErrors: true, useDefaults: true, keywords: ["uniforms"] });
  const schema = {
    title: 'Address',
    type: 'object',
    properties: {
     aa : {
        type: 'array',
        items: { type: 'object',
        properties: {
         city: { type: 'string' },
         state: { type: 'string' },
         street: { type: 'string' },
         option: { type: 'string', uniforms: { options: [
          { label: 'Option A', value: 'a' },
          { label: 'Option B', value: 'b' },
        ]} },
     }
    }
      },
    },
  };

  function createValidator(schema) {
    const validator = ajv.compile(schema);

    return (model) => {
      validator(model);

      if (validator.errors && validator.errors.length) {
        return { details: validator.errors };
      }
    };
  }

  const schemaValidator = createValidator(schema);

  return new JSONSchemaBridge(schema, schemaValidator);
})()
```

but it seems it cant use any custom component, is there any way so i cant apply my custom component in this schema ?

so the problem 
expected result : use custom component (multi select from antd)
<img width="920" alt="Screen Shot 2022-12-02 at 15 10 35" src="https://user-images.githubusercontent.com/50243317/205249069-3d7ca78d-439f-4bbc-95d5-5b9ac33371e8.png">

current result : use select component
<img width="643" alt="Screen Shot 2022-12-02 at 15 11 20" src="https://user-images.githubusercontent.com/50243317/205249110-c73dac9a-9c79-48b5-ab7b-839840a302c9.png">

**Comment by radekmie** [comment, marked as answer]
Hi @Zetra416. There are two things here:
1. To have a multi-select, your schema has to accept multiple values. That means, it has to be an array of strings, not a single string:
    ```ts
    option: {
      type: "array",
      uniforms: {
        options: [
          { label: "Option A", value: "a" },
          { label: "Option B", value: "b" },
        ],
      },
      items: { type: "string" },
    },
    ```
2. The above code will use a multi-select out of the box. If you'd like to provide your own custom component, then [follow this guide](https://uniforms.tools/docs/uth-autofield-algorithm/#overriding-autofield).
